server/task_check_scheduler.go

```
@@ -177,8 +177,60 @@ func (s *TaskCheckScheduler) Register(taskType string, executor TaskCheckExecuto
 	s.executors[taskType] = executor
 }
 
+func (s *TaskCheckScheduler) shouldScheduleTimingTaskCheck(ctx context.Context, task *api.Task) bool {
```

Should add a comment to explain when we would return true

---

Also we should return the err instead of failing silently